### ISSUE TITLE
Add different public key algorithm

### ISSUE THREAD

**Issue by thomas-tran** [decentralized-identity/ion#211]
Is it possible to add other public key algorithms to the Did Document like ED25519 , X25519 key agreement or custom key provided by external services?

**Comment by csuwildcat** [comment]
You can add any public key JWK representation you want to the DID Document. See here for some options: https://w3c-ccg.github.io/ld-cryptosuite-registry/ - just remember that it needs to be one of the JWK variants.

**Comment by thomas-tran** [comment]
Thanks

**Comment by thomas-tran** [comment]
@csuwildcat 

> You can add any public key JWK representation you want to the DID Document. See here for some options: https://w3c-ccg.github.io/ld-cryptosuite-registry/ - just remember that it needs to be one of the JWK variants.

Does the public key need to be in the JWK format? For example, I would to add to the DidDocument the public key for service authentication but in multibase format or hex format as mentioned in the did core
https://www.w3.org/TR/did-core/#referring-to-verification-methods

![image](https://user-images.githubusercontent.com/5725085/123880914-fc7fa180-d986-11eb-83dc-d7591e104796.png)

**Comment by OR13** [comment]
yes, IIRC ION only supports "publicKeyJwk" / "JsonWebKey2020"...

here are some did:key examples which you can test, https://github.com/transmute-industries/did-core/blob/main/packages/did-core-tests/__fixtures__/w3c-did-testsuite/did-key-transmute.json
